### ISSUE TITLE
DDC does not handle dynamic calls on null correctly

### ISSUE THREAD

**Issue by askeksa-google** [dart-lang/sdk#32194]
After the fix for https://github.com/dart-lang/sdk/issues/32173 where all calls on dynamic are treated as unknown, even if the method exists on Object, some DDC tests fail with runtime errors when calling Object methods on null.

**Comment by askeksa-google** [comment]
See this patchset for the exact failures: https://dart-review.googlesource.com/c/sdk/+/41762/7..8

**Comment by eernstg** [comment]
Right, here's the outcome from that discussion: [dynamic-members.md](https://github.com/dart-lang/sdk/blob/master/docs/language/informal/dynamic-members.md).
